src/array.rs

```diff
@@ -129,10 +135,14 @@ impl<V, A, Rhs> Mul<Rhs> for TArr<V, A>
 where
     V: Mul<Rhs>,
     A: Mul<Rhs>,
+    Rhs: Copy,
 {
     type Output = TArr<Prod<V, Rhs>, Prod<A, Rhs>>;
-    fn mul(self, _: Rhs) -> Self::Output {
-        unsafe { ::core::mem::uninitialized() }
+    fn mul(self, rhs: Rhs) -> Self::Output {
+        TArr {
+            first:  self.first * rhs,
+            rest:   self.rest  * rhs,
+        }
     }
 }
```

Comment:
This use of `::core::mem::uninitialized()` was sound because `TArr` merely wrapped a `PhantomData` and was thus guaranteed to have zero size, _regardless_ of the particular types stored by it.

By changing `TArr`'s definition so that its inner types also exist in the value domain, this assumption no longer holds.

Either `V` and `A` must be bounded to be ZSTs, or `Rhs` must be marked `Copy`. I took the latter approach.

Reply:
This seems reasonable to me. While this is technically a breaking change, it is documented that `TArr` should only be used with types from this library, which are all `Copy`.